docs/usage.rst

```diff
@@ -82,6 +82,143 @@ Use the batch writer to take care of dynamodb writing retries etc...
 The ``batch_writer()`` can take a keyword argument of ``flush_amount`` which will change the desired flush amount and a keyword argument
 of ``on_exit_loop_sleep``. The ``on_exit_loop_sleep`` argument will add an async sleep in the flush loop when you exit the context manager.
 
+
+S3 Examples
+-----------
+
+Here are some examples of uploading, checking if an S3 object 
+exists and streaming a file from S3, serving via aiohttp.
+
+Upload
+------
+
+Here we upload from a file object and stream it from a file descriptor.
+
+.. code-block:: python3
+
+    async def upload(
+        suite: str,
+        release: str,
+        filename: str,
+        staging_path: Path,
+        bucket: str,
+        aws_secret_access_key: str,
+        aws_access_key_id: str,
+    ) -> str:
+        blob_s3_key = f"{suite}/{release}/{filename}"
+        if not staging_path.exists():
+            LOG.error(
+                f"Unable to upload {blob_s3_key} - "
+                + f"Staging file {staging_path} does not exist"
+            )
+            return ""
+
+        async with aioboto3.client(
+            "s3",
+            aws_secret_access_key=aws_secret_access_key_write,
+            aws_access_key_id=aws_access_key_id_write,
+        ) as s3:
+            try:
+                with staging_path.open("rb") as spfp:
+                    LOG.info(f"Uploading {blob_s3_key} to s3")
+                    await s3.upload_fileobj(spfp, bucket, blob_s3_key)
+                    LOG.info(f"Finished Uploading {blob_s3_key} to s3")
+            except Exception as e:
+                LOG.error(
+                    f"Unable to s3 upload {staging_path} to {blob_s3_key}: "
+                    + f"{e} ({type(e)})"
+                )
+                return ""
+
+        return f"s3://{blob_s3_key}"
+
+Object Exists
+-------------
+
+Here we check to see if an object already exists
+
+.. code-block:: python3
+
+    async def blob_exists(
+        suite: str,
+        release: str,
+        filename: str,
+        bucket: str,
+        aws_secret_access_key: str,
+        aws_access_key_id: str,
+    ) -> bool:
+        blob_s3_key = f"{suite}/{release}/{filename}"
+        async with aioboto3.client(
+            "s3",
+            aws_secret_access_key=aws_secret_access_key,
+            aws_access_key_id=aws_access_key_id,
+        ) as s3:
+            object_list = await s3.list_objects_v2(
```

Comment:
If your making a function to check if an object exists, use `client.head_object()`

Reply:
I don't follow - I'll remove this function since it seems I'm going about it a less optimal way.